core/src/Cache/Items/MsalAccessTokenCacheItem.cs

```diff
@@ -83,36 +84,39 @@ internal MsalAccessTokenCacheItem
         [DataMember(Name = "expires_on", IsRequired = true)]
         internal long ExpiresOnUnixTimestamp { get; set; }
 
+        [DataMember(Name = "extended_expires_on")]
+        internal long ExtendedExpiresOnUnixTimestamp { get; set; }
+
         [DataMember(Name = "user_assertion_hash")]
         public string UserAssertionHash { get; set; }
 
         [DataMember(Name = "access_token_type")]
         internal string TokenType { get; set; }
 
-        internal string Authority
-        {
-            get
-            {
-                return string.Format(CultureInfo.InvariantCulture, "https://{0}/{1}/", Environment, TenantId ?? "common");
-            }
-        }
+        internal string Authority => string.Format(CultureInfo.InvariantCulture, "https://{0}/{1}/", Environment, TenantId ?? "common");
+        internal SortedSet<string> ScopeSet => ScopeHelper.ConvertStringToLowercaseSortedSet(NormalizedScopes);
 
-        internal SortedSet<string> ScopeSet
+        internal DateTimeOffset ExpiresOn
         {
             get
             {
-                return ScopeHelper.ConvertStringToLowercaseSortedSet(NormalizedScopes);
+                DateTime dtDateTime = new DateTime(1970, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc);
+                return dtDateTime.AddSeconds(ExpiresOnUnixTimestamp).ToUniversalTime();
             }
         }
-        internal DateTimeOffset ExpiresOn
+
+        internal DateTimeOffset ExtendedExpiresOn
         {
             get
             {
                 DateTime dtDateTime = new DateTime(1970, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc);
-                return dtDateTime.AddSeconds(ExpiresOnUnixTimestamp).ToUniversalTime();
+                return dtDateTime.AddSeconds(ExtendedExpiresOnUnixTimestamp).ToUniversalTime();
             }
         }
 
+        [DataMember]
```

Comment:
>DataMember [](start = 9, length = 10)

should the Name also be specified for this thing to ensure schema (casing) compliance?

Reply:
Yeah, I don't really know.  There isn't a name on it in the ADAL version.  This is why I had asked Petro to review this since he's more up to speed on the cache than I am.

Reply:
No, only "extended_expires_on" is required

Reply:
So, I talked with Olga and we shouldn't be persisting this value to the cache at all.  It should be calculated in memory on the fly but not be part of the cache.  So I'm going to talk with folks offline to resolve how to fix this without persisting it.